src/collector.rs

```diff
@@ -1,63 +1,282 @@
+                        }
+                    } else {
+                        Self::FsPath(value.into())
+                    }
+                }
+            }
+        }
+    }
+
+    pub async fn get_contents(
+        &self,
+        file_type_hint: Option<FileType>,
+        skip_missing: bool,
+    ) -> Result<Vec<InputContent>> {
+        use Input::*;
+
+        match self {
+            // TODO: should skip_missing also affect URLs?
+            RemoteUrl(url) => Ok(vec![Self::url_contents(url).await?]),
+            FsGlob {
+                pattern,
+                ignore_case,
+            } => Ok(Self::glob_contents(pattern, *ignore_case).await?),
+            FsPath(path) => {
+                let content = Self::path_content(&path).await.with_context(|| {
+                    format!(
+                        "Failed to read file: `{}`",
+                        path.to_str().unwrap_or("<MALFORMED PATH>")
+                    )
+                });
+                match content {
+                    Ok(input_content) => Ok(vec![input_content]),
+                    Err(_) if skip_missing => Ok(vec![]),
+                    Err(arg) => Err(anyhow!(arg)),
+                }
+            }
+            Stdin => Ok(vec![Self::stdin_content(file_type_hint).await?]),
+            String(s) => Ok(vec![Self::string_content(s, file_type_hint)?]),
+        }
+    }
+
+    async fn url_contents(url: &Url) -> Result<InputContent> {
+        let res = reqwest::get(url.clone()).await?;
+        let content = res.text().await?;
+        let input_content = InputContent {
+            input: Input::RemoteUrl(url.clone()),
+            file_type: FileType::from(url.as_str()),
+            content,
+        };
+
+        Ok(input_content)
+    }
+
+    async fn glob_contents(path_glob: &str, ignore_case: bool) -> Result<Vec<InputContent>> {
+        let mut contents = vec![];
+        let glob_expanded = tilde(&path_glob);
+        let mut match_opts = glob::MatchOptions::new();
+
+        match_opts.case_sensitive = !ignore_case;
+
+        for entry in glob_with(&glob_expanded, match_opts)? {
+            match entry {
+                Ok(path) => {
+                    let content = Self::path_content(&path).await?;
+                    contents.push(content);
+                }
+                Err(e) => println!("{:?}", e),
+            }
+        }
+
+        Ok(contents)
+    }
+
+    async fn path_content<P: Into<PathBuf> + AsRef<Path>>(path: P) -> Result<InputContent> {
+        let input_content = InputContent {
+            file_type: FileType::from(path.as_ref()),
+            content: read_to_string(&path).await?,
+            input: Input::FsPath(path.into()),
+        };
+
+        Ok(input_content)
+    }
+
+    async fn stdin_content(file_type_hint: Option<FileType>) -> Result<InputContent> {
+        let mut content = String::new();
+        let mut stdin = stdin();
+        stdin.read_to_string(&mut content).await?;
```

Comment:
Guess eventually we could read line-by-line for `stdin` in case people use lychee in a long-running process to avoid blocking and buffering. The question is if we ever gonna hit this case.

Reply:
Nice idea. This would work perfectly with #55